### PR TITLE
Add log when Poco Exception

### LINKED ISSUE #6580: Invalid access: Can not convert empty value
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

daily run window function and the failed sql is  `select i, i, row_number()  over w from window_function_0 window w as (partition by i order by id, i) order by 1, 2, 3`, the full error info is `Error 1105: other error for mpp stream: From MPP<query:<query_ts:0, local_query_id:0, server_id:0, start_ts:438529181194125313>,task_id:1>: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = Invalid access: Can not convert empty value., e.what() = Invalid access`.

### PR DESCRIPTION
**Pull request by CalvinNeo**
Signed-off-by: CalvinNeo <calvinneo1995@gmail.com>

### What problem does this PR solve?

Issue Number: ref #6580

The `result` may be invalid in `SchemaDiff::deserialize`. Don't know why, just add some log here.
```
void SchemaDiff::deserialize(const String & data)
{
    Poco::JSON::Parser parser;
    Poco::Dynamic::Var result = parser.parse(data);

```

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by hehechen — APPROVED** [review]

File: dbms/src/TiDB/Schema/SchemaGetter.cpp [hunk 1]
```diff
@@ -157,6 +157,10 @@ void SchemaDiff::deserialize(const String & data)
 {
     Poco::JSON::Parser parser;
     Poco::Dynamic::Var result = parser.parse(data);
+    if (result.isEmpty())
+    {
+        LOG_WARNING(&Poco::Logger::get("SchemaDiff"), "The schema diff deserialize failed {}", data);
```
Comment (JaySon-Huang) [inline dbms/src/TiDB/Schema/SchemaGetter.cpp:162]: Let's throw an Exception here and let the SchemaSyncService retry instead of making tiflash crash.

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**